lib/remote/apilistener.hpp

```diff
@@ -71,6 +71,22 @@ enum class ApiCapabilities : uint_fast64_t
 	IfwApiCheckCommand = 1u << 1u,
 };
 
+class ConfigSyncRule
+{
+public:
+	void WaitForObject(const Type::Ptr& ptype, const String& objName);
+
+	void BlockObject(const Type::Ptr& ptype, const String& objName);
+	void UnBlockNotify(const Type::Ptr& ptype, const String& objName, bool notifyAll = false);
+
+private:
+	bool IsObjectBlocked(const Type::Ptr& ptype, const String& objName);
+
+	std::mutex m_Mutex;
+	std::condition_variable m_CV;
+	std::unordered_map<Type*, std::set<String>> m_Objects;
```

Comment:
Maybe rename to `m_Locked` or maybe even `m_LockedObjectNames` to make more clear that being referenced in this map+set means that it's currently locked.